src/main/java/com/tractionrec/recrec/ui/RecRecRunning.java

```diff
@@ -28,22 +43,24 @@ public class RecRecRunning extends RecRecForm {
     private JButton nextButton;
     private List<Future<QueryResult>> futureResults;
 
+    private static final int MAX_CONCURRENT_REQUESTS = 50;
+    private final Semaphore requestSemaphore = new Semaphore(MAX_CONCURRENT_REQUESTS, false);
```

Comment:
without this virtual thread go crazy and try to run HTTP requests basically all at once, which overwhelms most machines - and probably annoys worldpay too